source/common/network/connection_impl.cc

```diff
@@ -655,7 +655,7 @@ void ConnectionImpl::onWriteReady() {
     int error;
     socklen_t error_size = sizeof(error);
     RELEASE_ASSERT(socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size).rc_ == 0,
```

Comment:
Could we pull the getSocketOption call out of the RELEASE_ASSERT and format the message with result.err_ since that would be the cross platform equivalent of grabbing errno after the sys call?